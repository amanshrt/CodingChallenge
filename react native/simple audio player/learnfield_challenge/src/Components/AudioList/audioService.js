

export const selectAudio = (data, navigation) => {
  try {
    navigation.navigate("Detail", { ...data })
    audioAction(data, 'PLAY');
  return true;
  }
  catch (e) { 
    return false
  }
};

export const audioAction = (data, event) => {
  console.log(data, event);
};
