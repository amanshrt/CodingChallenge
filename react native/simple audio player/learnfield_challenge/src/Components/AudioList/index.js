import React from 'react';
import {
  StyleSheet,
  Text,
  ImageBackground,
  TouchableOpacity, 
} from 'react-native';
import CommonStyles from '../../Assets/commonCss';
import LinearGradient from "react-native-linear-gradient";
import { selectAudio } from "./audioService";
import { useNavigation } from '@react-navigation/native';


const Section = ({ data }) => {
  const navigation = useNavigation();
  return (
    <TouchableOpacity style={CommonStyles.tinyLogo} onPress={()=> selectAudio(data ,navigation) }>
      <ImageBackground style={CommonStyles.image} source={{uri: data.cover}}>
        <LinearGradient
          colors={['#000000', '#00000000']}
          style={CommonStyles.linearGradient}></LinearGradient>
        <Text
          style={(CommonStyles.txt_bl, CommonStyles.txt_sm, styles.overlay)}>
          {data.totalDurationMs + 'ms'}
        </Text>
      </ImageBackground>
      <Text testID='data-render'
        style={
          (CommonStyles.txt_bl, CommonStyles.txt_lg, CommonStyles.txt_center)
        }>
        {data.title}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  overlay: {
    position : "absolute",
    right: 10,
  }
});

export default Section;
