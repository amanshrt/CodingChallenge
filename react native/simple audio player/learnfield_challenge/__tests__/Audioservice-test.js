import React from 'react';
import renderer from 'react-test-renderer';
import Section from "./../src/Components/AudioList/index";
import { selectAudio } from "./../src/Components/AudioList/audioService";
import { useNavigation } from '@react-navigation/native';
import LinearGradient from 'react-native-linear-gradient';
import SoundPlayer from 'react-native-sound-player';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});
jest.mock('react-native-sound-player', () => {});

test('renders correctly', () => {
  var data = {
    title: 'abc',
    cover: null,
    totalDurationMs: 1,
  };
  const tree = renderer.create(<Section data={data} />).toJSON();
  // expect(text).toHaveTextContent("data.title")
  //Another way to match snapshot
  expect(tree).toMatchSnapshot();
});


describe('Select the audio file and navigate to the screen 2', () => {
  test('passing audio file and navigation props', () => {
     var data = {
       title: 'abc',
       cover: null,
       totalDurationMs: 1,
     };
     const navigation = {
       navigate: jest.fn(),
     };
    const res = selectAudio(data, navigation);
    expect(res).toBe(true);
  });
});