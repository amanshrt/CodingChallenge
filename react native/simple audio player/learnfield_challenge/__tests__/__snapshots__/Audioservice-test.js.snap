// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#444",
      "flex": 1,
      "flexDirection": "column",
      "height": 220,
      "marginBottom": 12,
      "opacity": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": 180,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": null,
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 180,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <BVLinearGradient
      colors={
        Array [
          4278190080,
          0,
        ]
      }
      endPoint={
        Object {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={null}
      startPoint={
        Object {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "position": "absolute",
          "right": 10,
        }
      }
    >
      1ms
    </Text>
  </View>
  <Text
    style={
      Object {
        "textAlign": "center",
      }
    }
    testID="data-render"
  >
    abc
  </Text>
</View>
`;
